service/nodes.go

```diff
@@ -182,3 +240,82 @@ func (n *connectorNode) build(
 	}
 	return err
 }
+
+var _ consumerNode = &capabilitiesNode{}
+
+// Every pipeline has a "virtual" capabilities node immediately after the receiver(s).
+// There are two purposes for this node:
+// 1. Present aggregated capabilities to receivers, such as whether the pipeline mutates data.
+// 2. Present a consistent "first consumer" for each pipeline.
+// The nodeID is derived from "pipeline ID".
+type capabilitiesNode struct {
+	nodeID
+	pipelineID component.ID
+	baseConsumer
+	consumer.Capabilities
+}
+
+func newCapabilitiesNode(pipelineID component.ID) *capabilitiesNode {
+	return &capabilitiesNode{
+		nodeID:       newNodeID(capabilitiesSeed, pipelineID.String()),
+		pipelineID:   pipelineID,
+		Capabilities: consumer.Capabilities{},
+	}
+}
+
+func (n *capabilitiesNode) getConsumer() baseConsumer {
+	return n.baseConsumer
+}
+
+func (n *capabilitiesNode) build(next baseConsumer, processors []*processorNode) {
```

Comment:
I think this logic is wrong since you don't construct a different baseConsumer some time.

Reply:
I don't understand what's wrong. The node always passes data directly to the next consumer, so its baseConsumer is the same as the next consumer.

Reply:
Then why do you create them?

Reply:
Asked you a separate question, maybe that clarifies.

Reply:
Hopefully explained well here: https://github.com/open-telemetry/opentelemetry-collector/pull/7045#discussion_r1095108288